From 875d12e4f2e0ea885921f120b77a4a566f1a94d9 Mon Sep 17 00:00:00 2001
From: Song.Li <Song.Li@windriver.com>
Date: Mon, 3 Dec 2012 16:46:08 +0800
Subject: [PATCH] remove some checks for cross-compiling

some lib check is not suitable for
cross-compiling, so remove them.

Upstream-Status: Inappropriate [not a real bug,just for cross-compiling]

Signed-off-by: Song.Li <Song.Li@windriver.com>


Update patch for version 09.03.0300.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
--- psqlodbc-09.03.0300/configure.ac.orig	2014-08-06 13:57:56.511969275 +0800
+++ psqlodbc-09.03.0300/configure.ac	2014-08-06 14:03:39.595978624 +0800
@@ -48,57 +48,20 @@
 if test "$with_iodbc" != no; then
 	with_unixodbc=no
 	AC_DEFINE(WITH_IODBC, 1, [Define to 1 to build with iODBC support])
-	if test "$with_iodbc" = yes; then
-		AC_PATH_PROGS(ODBC_CONFIG, iodbc-config)
-	else
-		ODBC_CONFIG=$with_iodbc
-	fi
-	if test ! -f "${ODBC_CONFIG}/bin/iodbc-config"; then
-		if test ! -f "${ODBC_CONFIG}"; then
-			AC_MSG_ERROR([iodbc-config not found (required for iODBC build)])
-		fi
-	else
-		ODBC_CONFIG=${ODBC_CONFIG}/bin/iodbc-config
-	fi
 fi
 
 if test "$with_unixodbc" != no; then
 	AC_DEFINE(WITH_UNIXODBC, 1,
             [Define to 1 to build with unixODBC support])
-	if test "$with_unixodbc" = yes; then
-		AC_PATH_PROGS(ODBC_CONFIG, odbc_config)
-	else
-		ODBC_CONFIG=$with_unixodbc
-	fi
-	if test ! -f "${ODBC_CONFIG}/bin/odbc_config"; then
-		if test ! -f "${ODBC_CONFIG}"; then
-			AC_MSG_ERROR([odbc_config not found (required for unixODBC build)])
-		fi
-	else
-		ODBC_CONFIG=${ODBC_CONFIG}/bin/odbc_config
-	fi
 fi
 
 #
 # ODBC include and library
 #
 
-if test "$ODBC_CONFIG" != ""; then
-	if test "$with_iodbc" != no; then
-		ODBC_INCLUDE=`${ODBC_CONFIG} --cflags`
-		CPPFLAGS="$CPPFLAGS ${ODBC_INCLUDE}"
-		# Linking libiodoc is rather problematic
-		[ODBC_LIBDIR=`${ODBC_CONFIG} --libs | sed -e "s/^\(-L\|.*[ \t]-L\)\([^ \n\r\f\t]*\).*$/-L\2/"`]
-		LDFLAGS="$LDFLAGS ${ODBC_LIBDIR}"
-	else
-		ODBC_INCLUDE=`${ODBC_CONFIG} --include-prefix`
-		CPPFLAGS="$CPPFLAGS -I${ODBC_INCLUDE}"
-		# Linking libodoc is rather problematic
-		ODBC_LIBDIR=`${ODBC_CONFIG} --lib-prefix`
-		LDFLAGS="$LDFLAGS -L${ODBC_LIBDIR}"
-	fi
-	AC_MSG_NOTICE([using $ODBC_INCLUDE $ODBC_LIBDIR])
-fi
+ODBC_LIBS="-lodbcinst"
+LIBS="$LIBS ${ODBC_LIBS}"
+AC_MSG_NOTICE([using $ODBC_INCLUDE $ODBC_LIBS])
 
 #
 # SQLCOLATTRIBUTE_SQLLEN check
@@ -215,16 +178,10 @@
 #
 
 if test "$with_libpq" != no; then
-  if test -z "$PG_CONFIG"; then
-    AC_PATH_PROGS(PG_CONFIG, pg_config)
-  fi
-
-  if test -n "$PG_CONFIG"; then
-    pg_includedir=`"$PG_CONFIG" --includedir`
-    pg_libdir=`"$PG_CONFIG" --libdir`
-    CPPFLAGS="$CPPFLAGS -I$pg_includedir"
-    LDFLAGS="$LDFLAGS -L$pg_libdir"
-  fi
+  pg_includedir=""
+  pg_libdir=""
+  CPPFLAGS="$CPPFLAGS"
+  LDFLAGS="$LDFLAGS"
 fi
 
 
@@ -249,21 +206,11 @@
                  [AC_MSG_ERROR([iODBC library "iodbcinst" not found])])
 fi
 
-if test "$enable_pthreads" = yes; then
-  AC_CHECK_LIB(pthreads, pthread_create,
-               [],
-	       [AC_CHECK_LIB(pthread, pthread_create)])
-fi
-
 if test "$with_libpq" != no; then
   AC_CHECK_LIB(pq, PQconnectdb, [],
 	      [AC_MSG_ERROR([libpq library not found])])
 fi
 
-if test "$enable_openssl" = yes; then
-  AC_CHECK_LIB(ssl, SSL_read, [], [AC_MSG_ERROR([ssl library not found])])
-fi
-
 if test "$enable_gss" = yes; then
   AC_SEARCH_LIBS(gss_init_sec_context, [gssapi_krb5 gss 'gssapi -lkrb5 -lcrypto'], [],  [AC_MSG_ERROR([gssapi library not found])])
 fi
